src/AppInstallerCommonCore/Experiment.cpp

```diff
@@ -0,0 +1,109 @@
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "winget/Experiment.h"
+#include "winget/UserSettings.h"
+#include "Experiment/Experiment.h"
+#include "AppInstallerTelemetry.h"
+
+namespace AppInstaller::Settings
+{
+    namespace
+    {
+        ExperimentState GetExperimentStateInternal(Experiment::Key key, const UserSettings& userSettings)
+        {
+            if (key == Experiment::Key::None)
+            {
+                return { true, ExperimentToggleSource::Default };
+            }
+
+            if (!GroupPolicies().IsEnabled(TogglePolicy::Policy::Experiments))
+            {
+                AICLI_LOG(Core, Info, <<
+                    "Experiment " << Experiment::GetExperiment(key).Name() <<
+                    " is disabled due to group policy: " << TogglePolicy::GetPolicy(TogglePolicy::Policy::Experiments).RegValueName());
+                return { false, ExperimentToggleSource::Policy };
+            }
+
+            auto experiments = userSettings.Get<Setting::Experiments>();
+            auto experiment = Experiment::GetExperiment(key);
+            auto jsonName = std::string(experiment.JsonName());
+            if (experiments.find(jsonName) != experiments.end())
+            {
+                auto isEnabled = experiments[jsonName];
+                AICLI_LOG(Core, Info, <<
+                    "Experiment " << Experiment::GetExperiment(key).Name() <<
+                    " is set to " << isEnabled << " in user settings");

```

Comment:
I think this will log as 0 or 1 (unless we have `<<` overriden for bool?). Is that what you want?